client/my-sites/plans-features-main/index.tsx

```diff
@@ -295,15 +301,27 @@ const PlansFeaturesMain = ( {
 			}
 		}
 
+		const selectedStorageOption =
+			selectedStorageOptions && selectedStorageOptions[ cartItemForPlan?.product_slug || '' ];
+		const storageAddOn = storageAddOns?.find( ( addOn ) => {
+			return addOn?.featureSlugs?.includes( selectedStorageOption || '' );
+		} );
+
+		const storageAddOnItemForPlan = storageAddOn && {
+			product_slug: storageAddOn.productSlug,
+			quantity: storageAddOn.quantity,
+			volume: 1,
+		};
+
 		if ( onUpgradeClick ) {
-			onUpgradeClick( cartItemForPlan );
+			onUpgradeClick( cartItemForPlan, storageAddOnItemForPlan );
 			return;
 		}
 
 		const planPath = cartItemForPlan?.product_slug
 			? getPlanPath( cartItemForPlan.product_slug )
 			: '';
-		const checkoutUrlWithArgs = `/checkout/${ siteSlug }/${ planPath }`;
+		const checkoutUrlWithArgs = `/checkout/${ siteSlug }/${ planPath },${ storageAddOn?.productSlug }:-q-${ storageAddOn?.quantity }`;
```

Comment:
**Note:**

This change adds storage add-ons to the checkout cart for the `/plans` page